src/routes/users/fetch-users.e2e.test.js

```diff
@@ -0,0 +1,57 @@
+const { expect } = require('chai')
+const { fixtures, setupDB } = require('../../../test/utils')
+
+describe('GET /users', () => {   
+	describe('Failure', ()=> {
+		let userUser
+		let accessToken
+		before(async () =>{
+			userUser = await fixtures.insertUser({
+				role : 'user'
+			}) 
+			const body = { id: userUser.id, email: userUser.email }
+			accessToken = fixtures.generateAccessToken(
+				{user : body}
+			)
+		})
+		it('should return 403 if user is not an admin', async() =>{
+			const expectedError = {
+				"status":"failed",
+				"message":"Only admin can create users"
+			}
+			fixtures.api()
+				.get(`/api/v1/users`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.expect(403 , expectedError)
+		})
+	}) 
+	describe('Success', () => {
+		let adminUser
+		let accessToken
+		before(async () =>{
+			await setupDB()
```

Comment:
just curiosity at my end since I just noticed we don't use this function in the `before` block of our tests so far, just once before all the tests [here](https://github.com/tbf-challenge/teamwork-be/blob/main/test/pretest.js)

any reason why we need it here in these scenarios?

Reply:
we need to clear the db so no data inserted by the other tests interferes with this particular test

Reply:
okay, so which kind of data could interfere with the test?

Reply:
@beblicarl pending question

Reply:
1. Create comment test we insert multiple posts in the db
2. Like post test we insert multiple posts in the db
3. Flag post test we insert multiple posts in the db
4. All of them interfere with the fetch post process
5. Below, is the error message i get when i refuse to clear the db before the test

```
  AssertionError: expected 16 to deeply equal 6
      + expected - actual

      -16
      +6
```

Reply:
okay a bit clearer but you mention posts. thought this is a fetch users endpoint so not sure how it relates to posts

Reply:
sorry, i meant users.
infact, every single test we have inserts` users` to the db

Reply:
sure makes sense, i think tearing down the database works but might be quite expensive since it has to rerun the migrations and all. Let's focus on just clearing the user table

we can have a utils function, perhaps `resetDBTable` or `clearDBTable` or something better, it will take the table's name and clear just that data

I believe the tests should work fine that way

Reply:
yes please